### DISCUSSION TITLE
Istiod breaking when I include prometheus node-exporter in istio ambient mesh

### DISCUSSION THREAD

**Discussion by damieva** [istio/istio#51701] [Q&A]
Hello guys!
**Problem description**
As per the title, my implementation of istio is using the verion 1.22.1 in ambient mode and as platform I'm using a K8S cluster in version 1.28 running in EKS. Bassically the solution is working fine until I add the istio tag in prometheus node-exporter daemon set, after that the istiod pods stop working showing healthcheck errors and then ztunnel are also not working and finally the K8s nodes go down too.

**Event details**
Here the events of istiod when the problem start happend:
`  Warning  Unhealthy  3m48s  kubelet  Readiness probe failed: Get "http://10.0.92.193:8080/ready": dial tcp 10.0.92.193:8080: connect: connection refused
  Warning  Unhealthy  20s    kubelet  Readiness probe failed: Get "http://10.0.92.193:8080/ready": read tcp 10.0.92.118:44514->10.0.92.193:8080: read: connection reset by peer
  Warning  Unhealthy  17s    kubelet  Readiness probe failed: Get "http://10.0.92.193:8080/ready": read tcp 10.0.92.118:44520->10.0.92.193:8080: read: connection reset by peer
  Warning  Unhealthy  14s    kubelet  Readiness probe failed: Get "http://10.0.92.193:8080/ready": read tcp 10.0.92.118:53326->10.0.92.193:8080: read: connection reset by peer
  Warning  Unhealthy  14s    kubelet  Readiness probe failed: Get "http://10.0.92.193:8080/ready": EOF
  Warning  Unhealthy  11s    kubelet  Readiness probe failed: Get "http://10.0.92.193:8080/ready": read tcp 10.0.92.118:53334->10.0.92.193:8080: read: connection reset by peer
  Warning  Unhealthy  8s     kubelet  Readiness probe failed: Get "http://10.0.92.193:8080/ready": read tcp 10.0.92.118:53342->10.0.92.193:8080: read: connection reset by peer
  Warning  Unhealthy  5s     kubelet  Readiness probe failed: Get "http://10.0.92.193:8080/ready": read tcp 10.0.92.118:53344->10.0.92.193:8080: read: connection reset by peer
  Warning  Unhealthy  2s     kubelet  Readiness probe failed: Get "http://10.0.92.193:8080/ready": read tcp 10.0.92.118:60144->10.0.92.193:8080: read: connection reset by peer`
  
  
Then I tried to recreate istiod pods and I'm getting the following events:
`Events:
  Type     Reason                  Age   From               Message
  Normal   Scheduled               21s   default-scheduler  Successfully assigned istio-system/istiod-5546d766-9vqhk to ip-10-0-92-118.eu-west-2.compute.internal
  Warning  FailedCreatePodSandBox  20s   kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "b53f82d4f4a9e3e6cd9eac64e732de1806d6a0dce636f3b75ed7498594520a95": plugin type="aws-cni" name="aws-cni" failed (add): add cmd: Error received from AddNetwork gRPC call: rpc error: code = Unknown desc = RequestError: send request failed
caused by: Get "http://169.254.169.254/latest/meta-data/network/interfaces/macs/06:2d:e5:6c:6b:05/vpc-ipv4-cidr-blocks": read tcp 10.0.92.118:55538->169.254.169.254:80: read: connection reset by peer`

Also I tried to disable the HTTP probes rewrite but still not luck.
Any clue about what could be happening??
Thank you in advance.

**Comment by howardjohn** [comment]
This is a bug. Thanks for the report. I have sent a fix: https://github.com/istio/istio/pull/51711. Hopefully will merge quickly then be included in 1.22.2. It is because it is hostNetwork=true btw.

Note the host is pretty much broken at this point. There would need to be a cleanup of all the various networking modifications Istio made. I would recommend just getting new nodes if possible.

**Reply by damieva** [reply]
Thank you so much!!
